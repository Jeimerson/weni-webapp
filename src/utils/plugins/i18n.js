import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  en: en,
  'pt-br': pt_br,
  'pt-BR': pt_br,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
