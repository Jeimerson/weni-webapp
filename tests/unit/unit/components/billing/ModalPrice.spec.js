import { shallowMount, createLocalVue } from '@vue/test-utils';
import ModalPrice from '@/components/billing/ModalPrice.vue';
import i18n from '@/utils/plugins/i18n';

import Vuex from 'vuex';
import { org } from '../../../__mocks__';
const localVue = createLocalVue();
localVue.use(Vuex);

describe('ModalPrice.vue', () => {
  let wrapper;

  let actions;
  let store;
  let getters;
  let state;

  beforeEach(() => {
    actions = {};
    state = {
      BillingSteps: {
        billing_details: {
          address: {
            country: 'BR',
            state: 'AL',
          },
        },
      },
    };
    getters = {
      currentOrg() {
        return org;
      },
    };
    store = new Vuex.Store({
      actions,
      getters,
      state,
    });
    wrapper = shallowMount(ModalPrice, {
      localVue,
      i18n,
      propsData: {
        ranges: [],
      },
      store,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        unnnicIconSvg: true,
        unnnicSlider: true,
        unnnicTable: true,
        unnnicTableRow: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
