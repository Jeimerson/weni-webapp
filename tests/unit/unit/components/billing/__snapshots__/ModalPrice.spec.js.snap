// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPrice.vue renders a snapshot 1`] = `
<div class="billing-modal">
  <div class="billing-modal__content unnnic-grid-xl">
    <div class="unnnic-grid-span-12">
      <h1 class="billing-modal__content__title">Precificação</h1>
    </div>
    <div class="unnnic-grid-span-12">
      <unnnictable-stub items="" style="max-height: 280px; max-width: 800px;"></unnnictable-stub>
    </div>
    <div class="unnnic-grid-span-12 slider">
      <unnnicslider-stub step="1"></unnnicslider-stub>
      <p class="slider__title">Simulação de custo mensal</p>
      <h2 class="slider__value">US$ NaN</h2>
    </div>
    <div class="unnnic-grid-span-12 info">
      some specific text
    </div>
    <div class="close-button">
      <unnniciconsvg-stub icon="close-1" size="md" clickable=""></unnniciconsvg-stub>
    </div>
  </div>
</div>
`;
